### PR TITLE
Add test requirements

### PR DESCRIPTION
**Pull request by pganssle**
Switch over test dependency management to a more centralized `test_requirements.txt`.

HT @mariocj89

### CODE REVIEW

**Comment by jbrockmendel** [comment]
`six` is a regular requirement.  Are test_requirements on top of regular requirements or instead of?

**Comment by pganssle** [comment]
I suppose it doesn't matter much either way, since six is the *only* regular requirement.

**Comment by pganssle** [comment]
We can tweak the way requirements are being imported a bit later, for now I'll just leave `six`, because I'm trying to get us to a reasonable baseline so I can rebase the Isoparser branch.

File: tox.ini [hunk 1]
```diff
@@ -11,5 +11,4 @@ envlist =
 
 [testenv]
 commands = pytest {posargs}
-deps =
-    six,pytest,freezegun
+deps = -rtest_requirements.txt
```
Comment (mariocj89) [inline tox.ini]: I know I'm late here but you'd usually see requirements-dev.txt rather than test_requirements.

It is even "reserved" in PyPI https://pypi.python.org/pypi/requirements-dev.txt

Comment (pganssle) [reply tox.ini]: Definitely not too late to move it. What do you think of the suggestion of dropping `six` proposed above? We can do both changes in one PR.

Comment (mariocj89) [reply tox.ini]: Your requirement file should provide a "reproducible python environment". It should include all your dependencies.
What some people do if you want to split runtime/build vs test requirements you can include one from another.

Example: having `requirements.txt` with actual dependencies and `requirements-dev.txt` including requirements.txt (you can do -r)

Comment (mariocj89) [reply tox.ini]: What you might want though is to pin the versions in your requirements file.

Comment (mariocj89) [reply tox.ini]: You might be able to add mock using [markers](https://www.python.org/dev/peps/pep-0426/#environment-markers) as well here:
`mock ; python_version < '3.0'` (Try this out, I am just writing in github)

**Comment by jbrockmendel** [comment]
The six thing I asked about largely because I've had tox problems when a dependency gets specified multiple times.  If it isn't causing problems here, don't spend much time worrying about it.

File: setup.py [hunk 2]
```diff
@@ -24,7 +24,6 @@
       package_data={"dateutil.zoneinfo": ["dateutil-zoneinfo.tar.gz"]},
       zip_safe=True,
       requires=["six"],
-      tests_require=["freezegun", "pytest"],
```
Comment (wagner-certat) [inline setup.py]: Why did you remove this list? I could still be read from the file.

Comment (pganssle) [reply setup.py]: I believe this is only ever useful for `python setup.py test`, and now that we're using `pytest` (and haven't configured `setup.py test` to run the tests yet), it's useless to have it around. There's probably some utility in setting up `setup.py test` so it works properly, but my understanding is that the `test` target is considered something of a failed experiment.

Comment (wagner-certat) [reply setup.py]: > and now that we're using pytest (and haven't configured `setup.py` test to run the tests yet)

Ok, I can look up how other projects call pytest. I think it's quite simple.
